dask/dataframe/io/parquet/arrow.py

```diff
@@ -332,6 +339,65 @@ class ArrowDatasetEngine(Engine):
     # Public Class Methods
     #
 
+    @classmethod
+    def extract_filesystem(
+        cls,
+        urlpath,
+        filesystem,
+        dataset_options,
+        open_file_options,
+        storage_options,
+    ):
+
+        # Check if filesystem was specified as a dataset option
+        fs = dataset_options.pop("filesystem", None)
+        if filesystem is not None:
+            fs = filesystem
+
+        fs = fs or "fsspec"  # Default is fsspec
+
+        # Handle pyarrow-based filesystem
+        if isinstance(fs, pa_fs.FileSystem) or fs in ("arrow", "pyarrow"):
+            if isinstance(urlpath, (list, tuple, set)):
+                if not urlpath:
+                    raise ValueError("empty urlpath sequence")
+                urlpath = [stringify_path(u) for u in urlpath]
+            else:
+                urlpath = [stringify_path(urlpath)]
+
+            if fs in ("arrow", "pyarrow"):
+                fs = type(pa_fs.FileSystem.from_uri(urlpath[0])[0])(
+                    **(storage_options or {})
+                )
+
+            if isinstance(fs, pa_fs.FileSystem):
```

Comment:
I don't think the `isinstance` check is needed given the `if isinstance(fs, pa_fs.FileSystem) or fs in ("arrow", "pyarrow")` condition above 

```suggestion
            else:
```

Reply:
Are right - We don't need the else either